lib/src/main/java/org/esbtools/eventhandler/Promise.java

```diff
@@ -24,22 +24,24 @@
  * An asynchronous capture of potentially yet-to-come responses.
  *
  * <p>Allows code to build a {@link Future} object which will complete at some point in the future
- * with results from the {@link ResponsesHandler} passed to {@link #then(ResponsesHandler)}.
+ * with results from the {@link PromiseHandler} passed to {@link #then(PromiseHandler)}.
  *
  * @see Requester
  *
  * @param <T> The type of requests
- * @param <U> The type of responses
+ * @param <T> The type of responses
  */
-public interface ResponsePromise<T, U> {
+public interface Promise<T> extends Future<T> {
```

Comment:
This is bad IMO because you're conflating promises and futures. The promise should have no information on how to resolve itself, it should simply be way of representing callbacks for asynchronous operations.